table/block_based/block_based_table_builder.cc

```diff
@@ -1629,6 +1629,7 @@ void BlockBasedTableBuilder::WriteIndexBlock(
       }
       // The last index_block_handle will be for the partition index block
     }
+    s.PermitUncheckedError();
```

Comment:
Could you also just change the while loop:
    while (ok()) {
      Status s = rep_->index_builder->Finish(&index_blocks, *index_block_handle);
      if (!s.ok() && !s.IsIncomplete()) {
        rep_->SetStatus(s);
        return;
      }

I think (but have not tried it), that this would be equivalent

Reply:
Good question - I believe `while (ok() & s.IsIncomplete()) {` is still needed since `s.IsIncomplete()` is needed to indicate if we need to keep calling `rep_->index_builder->Finish`. So removing `s.IsIncomplete()` will make the case `s.ok() == true` case loops into `rep_->index_builder->Finish` when it shouldn't and cause test like `FormatVersions/BlockBasedTableTest.PartitionIndexTest/0` get stuck.

Reply:
Mark's suggestion doesn't break the loop on `s.ok()`

If you move `s.PermitUncheckedError()` to before the while loop, wouldn't that resolve the issue without being overly permissive?

Reply:
Yeah moving the `s.PermitUncheckedError()` before the loop also works - I moved it after so that it does not permit unchecking for `s.IsIncomplete()` as in "Usually I need to check for s.IsIncomplete but if I didn't check for s.IsIncomplete() due to table building being not ok(), then no need to check it by permitting uncheck".

But honestly I am fine with either way :)

Reply:
(Sorry GitHub didn't show me your comment as I was typing mine)

I know it smells like a case of one status superseding another, but IMHO it is not inherently so. Here's another proposal where there's only one potentially non-OK status at a time (assuming you treat `SetStatus` as a "move"):

```
bool finished = false;
while (ok() && !finished) {
  Status s = ...;
  if (s.ok()) {
    finished = true;
  } else if (!s.IsIncomplete()) {
    rep_->SetStatus(s);
    return;
  }
  ...
}
```

This or `s.PermitUncheckedError()` before the while is fine by me.

Reply:
Maybe even for clarity

```
if (s.ok()) {
  finished = true;
} else if (s.IsIncomplete()) {
  // More filters after this one
  assert(!finished);
} else {
  // Error
  rep_->SetStatus(s);
  return;
}
```

Reply:
Sounds good to me!